headers/types/ground_mode/ground_mode.h

```diff
@@ -774,4 +774,198 @@ struct swap_shop_inventory_data {
+    uint8_t field_0x13;
+    uint8_t field_0x14;
+    uint8_t field_0x15;
+    uint8_t field_0x16;
+    uint8_t field_0x17;
+};
+ASSERT_SIZE(struct ground_bg_sub_struct_1a0, 24);
+
+struct layer_specs {
+    int16_t num_tiles; // The number of tiles stored in the data + 1. The +1 is the null tile at the
+                       // beginning of tiles.
+    int16_t bpa_slot_num_tiles[4]; // The number of tiles in the BPA on this slot. 0 if no BPA is
+                                   // assigned.
+    int16_t num_chunks; // Number of chunks in the tilemap + 1. The +1 is the null chunk at the
+                        // beginning of tile mappings, that is not stored.
+};
+ASSERT_SIZE(struct layer_specs, 12);
+
+struct bpa_header {
+    uint8_t num_tiles;         // The number of individual tiles.
+    int16_t num_frames;        // The number of frames for each tile.
+    int duration_per_frame[0]; // Settings for each frame. One entry per frame.
+};
+
+struct bma_header {
+    // Map width/height that the camera in game will travel in tiles. Also the width/height of the
+    // collision and unknown data layers! For most maps (Map Width Chunks) * (Tiling Width) = (Map
+    // Width Camera).
+    uint8_t map_width_tiles;
+    uint8_t map_height_tiles;
+
+    // Map width/height in chunks. Also the width/height of the chunk mappings.
+    uint8_t map_width_chunks;
+    uint8_t map_height_chunks;
+
+    uint16_t num_layers; // Number of layers in this map. Must match BPC layer size. Allowed values
+                         // are only 1 or 2.
+    uint16_t has_data_layer; // Seems to be a boolean flag (0 or 1). If >0, the Unknown Data Layer
+                             // exists.
+    uint16_t has_collision;  // Number of Collision layers. 0, 1 or 2.
+};
+ASSERT_SIZE(struct bma_header, 10);
+
+struct bpl_header {
+    int16_t num_palettes;
+    int16_t has_pal_animations;
+};
+ASSERT_SIZE(struct bpl_header, 4);
+
+struct animation_specification {
+    int16_t duration_per_frame; // Time in game frames to hold a single palette frame for
+    int16_t num_frames; // Number of frames. This is also usually the length of frames in animation
+                        // palette, but it can also be less.
+};
+ASSERT_SIZE(struct animation_specification, 4);
+
+struct map_render {
+    int16_t chunk_dimensions;
+    int16_t field_0x2;
+    int16_t num_bgs;
+    bool wrap_around;
+    uint8_t field_0x7;
+    int width_chunks;
+    int height_chunks;
+    struct pixel_position map_size_pixels;
+    void (*tilemap_render_func)(struct map_render*);
+};
+ASSERT_SIZE(struct map_render, 28);
+
+struct ground_bg_sub_struct_2bc {
+    uint8_t field_0x0;
+    uint8_t field_0x1;
+    int16_t field_0x2;
+    int16_t num_layers;
+    int16_t field_0x6;
+    int16_t field_0x8;
+    int16_t field_0xa;
+    int16_t field_0xc;
+    int16_t field_0xe[2];
+    int16_t field_0x12;
+    int16_t field_0x14;
+    int16_t field_0x16;
+    void (*unk18)(void*, const void*, void*, int);
```

Comment:
This should have a better name

Reply:
It's not yet known what this function does, so I don't have a better name for it. Decomp makes it easier to figure out the function parameter arguments are without knowing conceptually what the function does, and this is one of those cases.